lib/grape/middleware/throttle_middleware.rb

```diff
@@ -11,6 +11,9 @@ def before
           period = 1.day
         elsif limit = throttle_options[:monthly]
           period = 1.month
+        elsif throttle_options[:period] && throttle_options[:limit]
```

Comment:
Looks good. Might avoid duplicate hash accesses with a slight tweak though. What do you think?

```ruby
elsif period = throttle_options[:period]
  limit = throttle_options[:limit]
```

I'm thinking it's good to also add a bit of error handling at this stage, since debugging might get ugly. Perhaps raise an exception if none of the acceptable params are specified.